crates/stages/src/stages/execution.rs

```diff
@@ -307,6 +307,14 @@ impl<EF: ExecutorFactory, DB: Database> Stage<DB> for ExecutionStage<EF> {
             tx.delete::<tables::StorageChangeSet>(key, None)?;
         }
 
+        // Look up the start index for the transaction range
+        let first_tx_num = tx.block_body_indices(*range.start())?.first_tx_num();
+
+        // Unwind all receipts for transactions in the block range
+        tx.unwind_table_by_num::<tables::Receipts>(first_tx_num)?;
+        // `unwind_table_by_num` doesn't unwind the provided key, so we need to unwind it manually
+        tx.delete::<tables::Receipts>(first_tx_num, None)?;
```

Comment:
it's better to compute `unwind_to_tx` target using `saturated_sub(1)` and provide it to `unwind_table_by_num`. no need to call `delete` then

Reply:
I think it won't work with the test we have and a hypothetical situation when `first_tx_num == 0`

Reply:
right